cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunTestsTask.java

```diff
@@ -175,19 +222,91 @@ public void execute(Project project) {
         JBallerinaBackend jBallerinaBackend = JBallerinaBackend.from(packageCompilation, JvmTarget.JAVA_17);
         JarResolver jarResolver = jBallerinaBackend.jarResolver();
+
+        // Only tests in packages are executed so default packages i.e. single bal files which has the package name
+        // as "." are ignored. This is to be consistent with the "bal test" command which only executes tests
+        // in packages.
+        runTestsUsingSuiteJSON(project, jarResolver, target, testsCachePath, jBallerinaBackend, cachesRoot);
+
+        // Cleanup temp cache for SingleFileProject
+        cleanTempCache(project, cachesRoot);
+        if (project.buildOptions().dumpBuildTime()) {
+            BuildTime.getInstance().testingExecutionDuration = System.currentTimeMillis() - start;
+        }
+    }
+
+    private void runTestsUsingSuiteJSON(Project project, JarResolver jarResolver, Target target, Path testsCachePath,
+                                        JBallerinaBackend jBallerinaBackend, Path cachesRoot) {
         TestProcessor testProcessor = new TestProcessor(jarResolver);
         List<String> moduleNamesList = new ArrayList<>();
         Map<String, TestSuite> testSuiteMap = new HashMap<>();
         List<String> updatedSingleExecTests;
-        // Only tests in packages are executed so default packages i.e. single bal files which has the package name
-        // as "." are ignored. This is to be consistent with the "bal test" command which only executes tests
-        // in packages.
         List<String> mockClassNames = new ArrayList<>();
+
+        boolean hasTests = createTestSuiteIfHasTests(project, target, testProcessor, testSuiteMap,
+                moduleNamesList, mockClassNames, this.isRerunTestExecution, this.report, this.coverage);
+
+        writeToTestSuiteJson(testSuiteMap, testsCachePath);
+
+        if (hasTests) {
+            int testResult;
+            try {
+                Set<String> exclusionClassList = new HashSet<>();
+                testResult = runTestSuite(target, project.currentPackage(), jBallerinaBackend, mockClassNames,
+                        exclusionClassList);
+
+                performTasksAfterTestCompletion(project, target, testsCachePath, jBallerinaBackend,
+                        cachesRoot, moduleNamesList, exclusionClassList);
+            } catch (IOException | InterruptedException | ClassNotFoundException e) {
+                cleanTempCache(project, cachesRoot);
+                throw createLauncherException("error occurred while running tests", e);
+            }
+
+            if (testResult != 0) {
+                cleanTempCache(project, cachesRoot);
+                throw createLauncherException("there are test failures");
+            }
+        } else {
+            out.println("\tNo tests found");
+        }
+    }
+
+    private void performTasksAfterTestCompletion(Project project, Target target, Path testsCachePath,
```

Comment:
```suggestion
    private void performPostTestsTasks(Project project, Target target, Path testsCachePath,
```

Reply:
Done in [4968e6e](https://github.com/ballerina-platform/ballerina-lang/pull/42183/commits/4968e6e4e65089855ddb48e35628b6b2d812786e) commit